mobly/controllers/android_device_lib/jsonrpc_client_base.py

```diff
@@ -218,8 +218,15 @@ def connect(self, uid=UNKNOWN_UID, cmd=JsonRpcCommand.INIT):
             ProtocolError: Raised when there is an error in the protocol.
         """
         self._counter = self._id_counter()
-        self._conn = socket.create_connection(('localhost', self.host_port),
-                                              _SOCKET_CONNECTION_TIMEOUT)
+        try:
+          self._conn = socket.create_connection(('localhost', self.host_port),
+                                                _SOCKET_CONNECTION_TIMEOUT)
+        except ConnectionRefusedError as err:
+          self.log.error('Failled to connect to localhost, trying 127.0.0.1: {}'
```

Comment:
Add an inline comment explaining what this retry covers (avoid referring to internal-specific info)